doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution = solveset(x**2 - y, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+### Put your equation into `Eq` form
+
+Put your equation into `Eq` form, then apply {func}`~.solveset` to the `Eq`. This approach is convenient if you are interactively solving an equation which cannot easily be rearranged to $expression = 0$.
+
+```py
+>>> from sympy import Eq, solveset
+>>> from sympy.abc import x, y
+>>> eqn = Eq(x**2, y)
+>>> solution = solveset(eqn, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+### Parse a string representing the equation
+Parse a string representing the equation into a form that SymPy can understand (`Eq` form), then apply {func}`~.solveset` to the parsed expression.  This approach is convenient if you are programmatically reading in a string. We [recommend against using parsing a string if you are creating the expression yourself](https://github.com/sympy/sympy/wiki/Idioms-and-Antipatterns#strings-as-input). Parsing an equation from a string requires you to use [transformations](https://docs.sympy.org/dev/modules/parsing.html?highlight=parse_expr#sympy.parsing.sympy_parser.parse_expr) for SymPy to handle equals signs and create symbols from your variables.
+
+You should always include the variable to solve for if you want to extract results programmatically, to ensure that SymPy solves for the desired variable.  
+    
+```py
+>>> from sympy import parse_expr, solveset
+>>> from sympy.abc import x
+>>> from sympy.parsing.sympy_parser import convert_equals_signs, standard_transformations
+>>> expr = "x ** 2 = y"
+>>> parsed = parse_expr(expr, transformations=standard_transformations + (convert_equals_signs,))
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solveset(parsed, x)
+>>> print(solutions)
+{-sqrt(y), sqrt(y)}
+```
+
+If you already have the equation in `Eq` form, you can parse that string:
+
+```py
+>>> from sympy import parse_expr, solveset
+>>> from sympy.abc import x
+>>> expr = "Eq(x**2, y)"
+>>> parsed = parse_expr(expr)
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solveset(parsed, x)
+>>> print(solutions)
+{-sqrt(y), sqrt(y)}
+```
+
+### {func}`~.solveset` can return infinitely many solutions when {func}`~.solve` cannot
+
+{func}`~.solveset` [can return infinitely many solutions](https://docs.sympy.org/dev/modules/solvers/solveset.html?highlight=solveset#why-solveset) and express them in standard mathematical notation, for example $\sin(x) = 0$ for $x = n * \pi$ for every integer value of $n$:
+
+```py
+>>> from sympy import pprint, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x)
+>>> pprint(solution)
+{2*n*pi | n in Integers} U {2*n*pi + pi | n in Integers}
+```
+
+However, {func}`~.solve` will return only a finite number of solutions (here, two):
+
+```py
+>>> from sympy import sin, solve
+>>> from sympy.abc import x
+>>> solution = solve(sin(x), x)
+>>> print(solution)
+[0, pi]
+```
```

Comment:
This should clarify why it is only 2 solutions and how that choice is made. Essentially the idea is that all solutions are generated from these two by adding multiples of the periodicity of the equation (here `2*pi`). The periodicity of other examples can be different:
```
In [18]: print(solve([sin(x)+sin(x/2)], [x], dict=True))
[{x: 0}, {x: -8*pi/3}, {x: -2*pi}, {x: -4*pi/3}, {x: 4*pi/3}, {x: 2*pi}, {x: 8*pi/3}, {x: 4*pi}]
```
(Not sure if that's really correct output)

Reply:
> Is this actually true in general?

Probably not but the I think the intention is for it to be true (CC @smichr).

Reply:
Stated why two, and that this is the goal.